Add Fastly configuration rst (#547)
FriendsOfSymfony/FOSHttpCache: doc/fastly-configuration.rst

.. _fastly configuration:

Fastly Configuration
-------------------

Caching is enabled by default. All requests that are marked as cacheable will be cached.

Tagging
~~~~~~~

Fastly supports :term:`tagging` out of the box.

> Requests can be tagged using a custom header named ``Surrogate-Keys``.

Reviewer: ```suggestion
Configure the tag header to ``Surrogate-Keys``. (``fos_http_cache.tags.response_header`` if you use FOSHttpCacheBundle)
```

Purge
~~~~~

Fastly supports two types of deletion of cached items using tags:

> 1. Soft Purge: This is the default behavior. It will mark the cached item as stale and will serve the stale item until the TTL expires. Once the TTL expires, the stale item will be evicted from the cache and the next request will be fetched from the origin.

Reviewer: not sure exactly, but usually soft means it "may" still be used when stale-while-revalidate is still allowed, but (at least with varnish) the data is still fetched from the backend when requested again. 

this phrase implies that soft purge would not have any effect :-)

```suggestion
1. Soft Purge: This is the default behavior. It will mark the cached item as stale. Stale items *may* still be served (e.g. with the ``stale-while-revalidate`` or ``stale-on-error`` cache control headers. Once the TTL expires, the stale item will be evicted from the cache and the next request must be fetched from the origin.
```

Reviewer: we could link "Soft purge" to the fastly purge documentation at https://docs.fastly.com/en/guides/soft-purges

Author: I think you have to read the documentation anyway. In my POV we "could" but it wouldn't really help to use the caching without reading the documentation of Fastly. ;-)

Reviewer: lets add a note to the top of this page then. linking to https://docs.fastly.com/ and saying that you should also study the documentation of fastly to set it up correctly. (its kind of obvious, but just to set expectations straight)

> 1. Soft Purge: This is the default behavior. It will mark the cached item as stale. Stale items *may* still be served (e.g. with the ``stale-while-revalidate`` or ``stale-on-error`` cache control headers. Once the TTL expires, the stale item will be evicted from the cache and the next request must be fetched from the origin.

Reviewer: ```suggestion
1. Soft Purge: This is the default behavior. It will mark the cached item as stale. Stale items *may* still be served (e.g. with the ``stale-while-revalidate`` or ``stale-on-error`` cache control headers). Once the TTL expires, the stale item will be evicted from the cache and the next request must be fetched from the origin.
```

2. Purge: This will immediately evict the cached item from the cache and the next request will be fetched from the origin.

There are different ways to purge the cache:

> 1. Using Tags

Reviewer: needs a blank line to have sphinx see this is an enumeration

```suggestion

1. Using Tags
```

2. Using URLs

Cache Header
~~~~~~~~~~~~

To specify how long items should be cached you can use the ``Surrogate-Control`` header.
Using this header enables you to use the ``Cache-Control`` header to specify the cache duration for browsers.

> Actually this package doesn't support the ``Surrogate-Control`` header.

Reviewer: this is something i hope to add soonish.

```suggestion
Currently, FOSHttpCache does not yet support the ``Surrogate-Control`` header.
```
